cpp/src/arrow/gpu/cuda_memory.h

```diff
@@ -151,25 +151,25 @@ class ARROW_EXPORT CudaIpcMemHandle {
 /// \class CudaBufferReader
 /// \brief File interface for zero-copy read from CUDA buffers
 ///
-/// Note: Reads return pointers to device memory. This means you must be
-/// careful using this interface with any Arrow code which may expect to be
-/// able to do anything other than pointer arithmetic on the returned buffers
+/// CAUTION: reading to a Buffer returns a Buffer pointing to device memory.
+/// It will generally not be compatible with Arrow code expecting an usual
```

Comment:
Hmm, I must admit I don't understand the suggestion, but perhaps it's because I'm not a native English speaker. Would "the usual CPU buffer" be some kind of idiomatic expression?

Reply:
That is idiomatic, but has the connotation of a specific CPU buffer (like there's some singleton to which you are returning a reference).

Reply:
Ah, well, I'm not talking about a specific CPU buffer here.

Reply:
Right, so the suggested change above was to avoid that implication

Reply:
I'm not sure I understand: you're saying "an usual buffer" implies a specific buffer, but "the usual buffer" doesn't? I would expect the reverse (but again, not a native English speaker :-)).

Reply:
Sorry, I was comparing my suggestion "the usual case of a CPU buffer" to "the usual CPU buffer" (which I thought you were suggesting above). The original ("an usual buffer") is not correct English

Reply:
Ah, thank you! I'll change to "the usual case of a CPU buffer" then.

Reply:
Actually, I'll change to "Arrow code expecting a buffer pointing to CPU memory".